### PR TITLE
Better traces in list operations.

### PR DESCRIPTION
**Pull request by wojtek-t**

### CODE REVIEW

File: pkg/apiserver/resthandler.go [hunk 1]
```diff
@@ -316,16 +319,23 @@ func ListResource(r rest.Lister, rw rest.Watcher, scope RequestScope, forceWatch
...
+		// Log only long List requests (ignore Watch).
+		defer trace.LogIfLong(500 * time.Millisecond)
+		trace.Step("About to List from storage")
 		result, err := r.List(ctx, &opts)
 		if err != nil {
 			errorJSON(err, scope.Codec, w)
 			return
 		}
-		if err := setListSelfLink(result, req, scope.Namer); err != nil {
+		trace.Step("Listing done")
```
Comment (gmarek) [inline pkg/apiserver/resthandler.go]: Make it more explicit: "Done listing from storage" or sth.

Comment (wojtek-t) [inline pkg/apiserver/resthandler.go]: Done.

**Comment by gmarek** [comment]
LGTM

**Comment by wojtek-t** [comment]
trivial rebase

**Comment by wojtek-t** [comment]
@k8s-bot e2e test this please

**Pull request merged**